internal/opencode/catalog_test.go

```
@@ -115,6 +157,55 @@ func TestDiscoverCatalogRejectsInvalidOutput(t *testing.T) {
 
+// TestProcessStreamReaderReadSurfacesExitStatusAfterEOF pins the Read
+// contract: once the drained stream is exhausted, Read must reap the child
+// and return its exit error, never swallow it as a plain io.EOF.
+func TestProcessStreamReaderReadSurfacesExitStatusAfterEOF(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("requires a POSIX exit-status fixture")
+	}
+	dir := t.TempDir()
+	helper := filepath.Join(dir, "exit-helper")
+	if runtime.GOOS == "windows" {
+		helper += ".exe"
```

golangci-lint (2.13.2)

[error] ineffectual assignment to helper

(ineffassign)